statistics/handle/handle_hist.go

```diff
@@ -147,70 +148,78 @@ func (h *Handle) SubLoadWorker(ctx sessionctx.Context, exit chan struct{}, exitW
 			}
 		}
 	}()
+	var lastTask *NeededColumnTask
 	for {
-		err := h.handleOneTask(readerCtx, ctx.(sqlexec.RestrictedSQLExecutor), exit)
+		task, err := h.HandleOneTask(lastTask, readerCtx, ctx.(sqlexec.RestrictedSQLExecutor), exit)
+		lastTask = task
 		if err != nil {
 			switch err {
 			case errExit:
 				return
 			default:
-				time.Sleep(10 * time.Millisecond)
+				time.Sleep(h.Lease() / 10)
 				continue
 			}
 		}
 	}
 }
 
-// handleOneTask handles one column task.
-func (h *Handle) handleOneTask(readerCtx *statsReaderContext, ctx sqlexec.RestrictedSQLExecutor, exit chan struct{}) (err error) {
+// HandleOneTask handles one column task.
+func (h *Handle) HandleOneTask(lastTask *NeededColumnTask, readerCtx *StatsReaderContext, ctx sqlexec.RestrictedSQLExecutor, exit chan struct{}) (task *NeededColumnTask, err error) {
```

Comment:
We can add some comments about the `lastTask`

Reply:
Will do it.

Reply:
Done